contrib/luasocket/premake5.lua

```diff
@@ -0,0 +1,32 @@
+project "luasocket-lib"
+	language    "C"
+	kind        "StaticLib"
+	warnings    "extra"
+	pic         "on"
+
+	includedirs { "../lua/src" }
+
+	files
+	{
+		"src/*.c",
+		"src/*.h",
+		"src/*.lua"
+	}
+
+	filter "system:windows"
+		removefiles
+		{
+			"src/serial.c",
+			"src/unixdgram.*",
+			"src/unixstream.*",
+			"src/unix.*",
+			"src/usocket.*",
+		}
+	
+	filter "system:not windows"
+		removefiles
+		{
+			"src/wsocket.*",
+		}
+		
+		defines { "LUASOCKET_API=__attribute__((visibility(\"default\")))" }
```

Comment:
Generally, files should end with an empty line - specifically an empty line, not a blank line with indentation. That's what this two red emojis are all about here. Apparently some tools in the "*nix" world assume text files end with an empty line, and searching for `LUASOCKET_API` would not flag this file due to that.

Reply:
Right, I'll fix this up with a newline then